feat: thirdparty rework (#522)
supertokens/docs: v2/thirdparty/reusableMD/multitenancy/backend-cors.mdx

> ---

Reviewer: use copy section instead

id: backend-cors
title: "Backend CORS"
hide_title: true
---

import BackendSDKTabs from "/src/components/tabs/BackendSDKTabs";
import GoFrameworkSubTabs from "/src/components/tabs/GoFrameworkSubTabs";
import TabItem from '@theme/TabItem';

> If you are using different domains or sub-domains for each tenant on the frontend, the corsMiddleware needs to updated to allow requests from them.

Reviewer: ```suggestion
If you are using different domains or sub-domains for each tenant on the frontend, your backend CORS setting needs to be updated to allow requests from those origins.
```

<BackendSDKTabs>
<TabItem value="nodejs">

```tsx
TODO
```
</TabItem>
<TabItem value="go">

<GoFrameworkSubTabs>

<TabItem value="http">

```go
import (
	"net/http"
	"strings"

	"github.com/supertokens/supertokens-golang/supertokens"
)

func corsMiddleware(next http.Handler) http.Handler {
	return http.HandlerFunc(func(response http.ResponseWriter, r *http.Request) {
		// highlight-start
		origin := r.Header.Get("Origin")
		if !validateOrigin(origin) {
			response.WriteHeader(422)
			response.Write([]byte("Invalid tenant"))
			return
		}

> 		response.Header().Set("Access-Control-Allow-Origin", origin)

Reviewer: change logic to add access control allow origin only if validateOrigin

		// highlight-end
		response.Header().Set("Access-Control-Allow-Credentials", "true")
		if r.Method == "OPTIONS" {
			// we add content-type + other headers used by SuperTokens
			response.Header().Set("Access-Control-Allow-Headers",
				strings.Join(append([]string{"Content-Type"},
					supertokens.GetAllCORSHeaders()...), ","))
			response.Header().Set("Access-Control-Allow-Methods", "*")
			response.Write([]byte(""))
		} else {
			next.ServeHTTP(response, r)
		}
	})
}

// highlight-start

> func validateOrigin(origin string) bool {

Reviewer: ```suggestion
func isAllowedOrigin(origin string) bool {
```

> 	// logic to validate origin
> 	return true

Reviewer: ```suggestion
	// TODO: Return true if any of the tenants in your app use the input origin.
	return true
```
